internal/controller/controller.go

```diff
@@ -220,6 +224,7 @@ func (r *reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 		reqLogger := logger.WithValues("organization_id", orgID, "request_id", requestID)
 		ctx = log.IntoContext(ctx, reqLogger)
 
+		r.removeConfiguredAttributes(reqLogger, obj)
```

Comment:
not necessary to pass logger to this function, it just logs an info message

Reply:
Do you mean that maybe we should just log above it to avoid passing the param? The trouble with that is we'd have to duplicate the guard clause that returns early if no removals are configured. I think passing the logger as a parameter is a small price to pay for that! I guess it'd be more idiomatic to pass the context instead and retrieve it from there?

Reply:
just a nit (I guess the only thing I miss in java is not passing loggers around)
```
logger.Info("removing configured resource attributes")
r.removeConfiguredAttributes(obj)
```

Reply:
But see my first reply for why I don't want to do that - it'd remove the distinction between redacted and unredacted types in the logs.

I know what you mean about passing a lot of cross-cutting parameters. The idiomatic go equivalent is to put and get values in the request context, as we do in some other places IIRC... I'll actually change it to that just to set an example 😂